packages/application-generic/src/usecases/add-job/add-job.usecase.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { Test } from '@nestjs/testing';
+import { ExecutionDetailsRepository, JobRepository } from '@novu/dal';
+import { UserSession } from '@novu/testing';
+
+import { AddJob, AddDigestJob, AddDelayJob } from './index';
+
+import { CreateExecutionDetails } from '../create-execution-details';
+
+import {
+  CalculateDelayService,
+  DistributedLockService,
+  EventsDistributedLockService,
+  InMemoryProviderService,
+  QueueService,
+} from '../../services';
+
+let inMemoryProviderService: InMemoryProviderService;
+
+describe('AddJob use case', function () {
+  let useCase: AddJob;
+
+  beforeEach(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [],
+      providers: [
+        {
+          provide: InMemoryProviderService,
+          useFactory: () => {
+            return new InMemoryProviderService();
+          },
+        },
+        CalculateDelayService,
+        DistributedLockService,
+        EventsDistributedLockService,
+        ExecutionDetailsRepository,
+        JobRepository,
+        CreateExecutionDetails,
+        AddDigestJob,
+        AddDelayJob,
+      ],
+    }).compile();
+
+    const jobRepository = moduleRef.get<JobRepository>(JobRepository);
+    const queueService = new QueueService();
+    const createExecutionDetails = moduleRef.get<CreateExecutionDetails>(
+      CreateExecutionDetails
+    );
+    const addDigestJob = moduleRef.get<AddDigestJob>(AddDigestJob);
+    const addDelayJob = moduleRef.get<AddDelayJob>(AddDelayJob);
+    useCase = new AddJob(
+      jobRepository,
+      queueService,
+      createExecutionDetails,
+      addDigestJob,
+      addDelayJob
+    );
+  });
+
+  afterAll(async () => {
+    await useCase.getQueueService().gracefulShutdown();
+  });
+
+  it('should instantiate properly the use case', async () => {
+    const queueService = useCase.getQueueService();
+
+    expect(queueService.DEFAULT_ATTEMPTS).toEqual(3);
+    expect(queueService.name).toEqual('standard');
+    const { bullMqService } = queueService;
+    expect(bullMqService).toBeDefined();
+    expect(await bullMqService.getRunningStatus()).toEqual({
+      queueIsPaused: false,
+      queueName: 'standard',
+      workerIsRunning: undefined,
+      workerName: undefined,
+    });
+  });
```

Comment:
You've created the test for the AddJob use-case but just checking if the queue is instantiated properly? 🤔 
is something missing here?

Reply:
I am trying to validate that the queueName used by the BullMqService is the right one for the intentions of AddJob. I am facing problems with the DI on the MemoryDB PR and I am going back and forth from `next` to that PR to double check I am doing the changes properly.

Reply:
If you are facing problems with DI typically it's because few provider instances have the same interface. If that's still a problem maybe you should use the "named" injections.